bluesky/tests/test_new_examples.py

```diff
@@ -807,3 +812,62 @@ def bad_sig(detectors, mtr, step):
     with pytest.raises(TypeError) as excinfo:
         RE(scan([hw.det], hw.motor, -1, 1, 3, per_step=bad_sig))
     assert excinfo.match("per_step must be a callable with the signature")
+
+
+def test_stage_all_and_unstage_all(RE):
+
+    from ophyd import StatusBase
+
+    staged = {}
+    unstaged = {}
+
+    # Test support for old and new style devices
+    class OldStyleDummy:
+        def __init__(self, name: str) -> None:
+            self.name: str = name
+
+        def stage(self):
+            staged[self.name] = True
+            ttime.sleep(0.5)
+            return [self]
+
+        def unstage(self):
+            unstaged[self.name] = True
+            ttime.sleep(0.5)
+            return [self]
+
+    class NewStyleDummy:
+        def __init__(self, name: str) -> None:
+            self.name: str = name
+
+        def _callback(self, d: dict, st: Status):
+            d[self.name] = True
+            st.set_finished()
+
+        def stage(self) -> Status:
+            st = StatusBase()
+            threading.Timer(0.5, self._callback, args=[staged, st]).start()
+            return st
+
+        def unstage(self) -> Status:
+            st = StatusBase()
+            threading.Timer(0.5, self._callback, args=[unstaged, st]).start()
+            return st
+
+    olddummy1 = OldStyleDummy("o1")
+    olddummy2 = OldStyleDummy("o2")
+    newdummy1 = NewStyleDummy("n1")
+    newdummy2 = NewStyleDummy("n2")
+
+    def plan():
+        yield from stage_all(olddummy1, olddummy2, newdummy1, newdummy2)
+        assert list(staged.keys()) == ["o1", "o2", "n1", "n2"]
+
+        yield from unstage_all(olddummy1, olddummy2, newdummy1, newdummy2)
+        assert list(unstaged.keys()) == ["o1", "o2", "n1", "n2"]
+
+    start = ttime.time()
+    RE(plan())
+    stop = ttime.time()
```

Comment:
`ttime.monotonic()` will be more stable to handle the DST-related changes (see https://docs.python.org/3/library/time.html#time.monotonic for details).
```suggestion
    start = ttime.monotonic()
    RE(plan())
    stop = ttime.monotonic()
```